overlord/snapshotstate/backend/backend.go

```diff
@@ -164,6 +179,25 @@ func Filename(snapshot *client.Snapshot) string {
 	return filepath.Join(dirs.SnapshotsDir, fmt.Sprintf("%d_%s_%s_%s.zip", snapshot.SetID, snapshot.Snap, snapshot.Version, snapshot.Revision))
 }
 
+func isSnapshotFilename(fname string) (ok bool, setID uint64) {
+	// XXX: we could use a regexp here to match very precisely all the elements
+	// of the filename following Filename() above, but perhaps it's better no to
+	// go overboard with it in case the format evolves in the future. Only check
+	// if the name starts with a set-id and ends with .zip.
+	if ext := filepath.Ext(fname); ext != ".zip" {
+		return false, 0
+	}
+	parts := strings.SplitN(fname, "_", 2)
```

Comment:
Nitpick, but this will work for `1_.zip` too. Maybe we should expect the parts[1] to conform to snap name? We could drop the extension and then exploy naming.ValidateSnap().

Reply:
Right. I'm not sure we should do that, but I'm open to discuss. FWTW we are mostly interested in filtering out non-zip files and files where we cannot get set-id from the beginning of the filename. The snapshots directory is fully managed and owned by snapd so we don't expect random files there, the general idea is to make it future-proof in case we want to add something new there. At the moment it is not even possible to add a subdirectory there without triggering errors during iteration ;)

Reply:
the point of this is mostly filtering out non zips, and finding the set id from the name.  I wouldn't got the full ValidateSnap route, OTOH it might make sense to ignore things where len(parts[1]) == 0, we expect something to be there